packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2891,7 +2906,7 @@ class _Location {
 /// [Dart Kernel Transformer](https://github.com/dart-lang/sdk/wiki/Kernel-Documentation).
 ///
 /// Currently creation locations are only available for [Widget] and [Element]
-_Location _getCreationLocation(Object object) {
+_Location getCreationLocation(Object object) {
```

Comment:
I exposed this so the error reporting can print the corresponding location

Reply:
Now that you expose isUserCreated do you need to expose this directly?
If it is needed, we should not return a private _Location object instance. Instead we would need to make _Location implement a public SourceLocation class with a more restricted API that we are willing to make public. In particular it shouldn't include the parameter locations which we may want to remove.

Reply:
will you suggest me return location instead of bool in isLocalCreationLocation? I will need the location info to print out the error message.

Reply:
ah yes I hadn't reviewed that part of the code yet when I made the comment.
Suggestion: expose an API in `widget_inspector.dart` that returns a DiagnosticsNode describing the location for an object as well as a short description of the object. That way we don't leak the private _Location class that we probably want to refactor.
E.g. the Api is something like
```dart
describeLocation(Diagnosticable object) {
  location = ...
  return ErrorDescription('${object.toDiagnosticsNode().toStringShort()} $location');
}
```

Reply:
Sounds good, I made it a string instead, since I will compose the information in a later stage.